### ISSUE TITLE
Grommet Accessibility Audit

### ISSUE THREAD

**Issue by taysea** [grommet/grommet#4068]
# Purpose of this Issue
The purpose of this issue is to identify keyboard/ARIA accessibility improvements that can be made in Grommet to better align the library with [WCAG standards](https://www.w3.org/TR/WCAG21/). Grommet does a great job of building in accessibility, but there are some areas for improvement.

## What is accessibility?
Accessibility is the practice of making your websites usable by as many people as possible. Some users can navigate only via keyboard, or screen reader, etc. You might also think of accessibility as treating everyone the same, and giving them equal opportunities, no matter what their ability or circumstances.

Source: https://developer.mozilla.org/en-US/docs/Learn/Accessibility/What_is_accessibility

## How were these improvements identified?
Grommet components were assessed for their keyboard navigability and success with Voiceover. Their behavior was compared side by side with the recommendations from WCAG. Typically, searching for [component] + wcag keyboard returns links to the appropriate section of w3 site.

For example, searching range "input wcag keyboard" returns this helpful resource as a top result: https://www.w3.org/TR/wai-aria-practices/examples/slider/slider-1.html

The behavior of the above slider example (via mouse, keyboard only, and screen reader) can be compared side by side with the Grommet component to identify any discrepancies. Additionally, there is often direct guidance for what the keyboard behavior should be.

Additionally, for some FormField label things, the WAVE chrome browser plugin was used. This identifies certain [ARIA](https://www.w3.org/TR/wai-aria-practices/) and color contrast errors.

## Grommet Component Analysis Results
The following are the areas of improvement that need to be address and have tests written for.

### FormField
- Providing a label on a FormField and on its contained Checkbox leads to an orphaned label on the formfield.

### Meter
- [ ] Similar to Chart, screen reader users should be able to identify what's in the chart without seeing it.
- Currently, Voiceover doesn't have access to read chart text information; just read out as "image" with no content information
- Recommendation, some sort of aria attribute should explain the value.

**Comment by ShimiSun** [comment]
Excellent analysis @taysea ! thank you for this contribution.
We will work on adding accessibility tests as well to enforce the best practices.

**Comment by eddieshowcase** [comment]
I would like to add the this list of items that need some review or attention:

### Tabs

- Tab components should respond to use of arrow keys for navigation (not just rely on 'tab' key)
- https://www.w3.org/TR/wai-aria-practices/examples/tabs/tabs-2/tabs.html

### Accordion

- Similar to Tabs, should respond to use of arrow keys for navigation of top-level nodes (not just rely on 'tab' key). Arrow keys should allow user to navigate Accordion items, then 'tab' key to navigate in/out of Accordion panel content.
- https://www.w3.org/TR/wai-aria-practices-1.1/examples/accordion/accordion.html

### Menu (add to bullets already mention)

- Beyond focusing first element upon opening menu, component should respond to use of arrow keys for navigation of items.
- https://www.w3.org/TR/wai-aria-practices/examples/menu-button/menu-button-links.html

### Sidebar / Navbar

- Sidebar or Navbar navigation components/examples should be implemented essentially as vertical Tabs and/or a Menu, and also respond to arrow-key navigation with appropriate aria role and not rely on 'tab' keep only.
- https://www.w3.org/TR/wai-aria-practices/examples/menubar/menubar-1/menubar-1.html
- This may not be baked into components, but at the very least documented in guidance on the design system website.


Thanks, y'all are awesome!

**Comment by jcfilben** [comment]
Closing this larger issue and opening seperate issues as needed for the things mentioned in this thread.

CheckBox accessibility issue can be tracked here:
https://github.com/grommet/grommet/issues/7325

Meter accessibility issue can be tracked here:
https://github.com/grommet/grommet/issues/7326

Tab, Accordion, and Sidebar, not opening a separate issue for these at the moment since they are not WCAG violations and the current pattern isn't broken. Can revisit this later.

Menu
Beyond focusing first element upon opening menu, component should respond to use of arrow keys for navigation of items.
This issue has been resolved.
